### PR TITLE
Fixed exception message when an annotation does not provide any properties

### PR DESCRIPTION
**Pull request by lyrixx**

### CODE REVIEW

File: lib/Doctrine/Common/Annotations/DocParser.php [hunk 1]
```diff
@@ -823,7 +823,11 @@ private function Annotation()
         foreach ($values as $property => $value) {
             if (!isset(self::$annotationMetadata[$name]['properties'][$property])) {
                 if ('value' !== $property) {
-                    throw AnnotationException::creationError(sprintf('The annotation @%s declared on %s does not have a property named "%s". Available properties: %s', $originalName, $this->context, $property, implode(', ', self::$annotationMetadata[$name]['properties'])));
+                    if (self::$annotationMetadata[$name]['properties']) {
+                        throw AnnotationException::creationError(sprintf('The annotation @%s declared on %s does not have a property named "%s". Available properties: %s.', $originalName, $this->context, $property, implode(', ', self::$annotationMetadata[$name]['properties'])));
+                    } else {
```
Comment (Ocramius) [inline lib/Doctrine/Common/Annotations/DocParser.php]: `else` is redundant - if it throws, no `else` is needed

File: lib/Doctrine/Common/Annotations/DocParser.php [hunk 2]
```diff
@@ -823,7 +823,11 @@ private function Annotation()
         foreach ($values as $property => $value) {
             if (!isset(self::$annotationMetadata[$name]['properties'][$property])) {
                 if ('value' !== $property) {
-                    throw AnnotationException::creationError(sprintf('The annotation @%s declared on %s does not have a property named "%s". Available properties: %s', $originalName, $this->context, $property, implode(', ', self::$annotationMetadata[$name]['properties'])));
+                    if (self::$annotationMetadata[$name]['properties']) {
```
Comment (Ocramius) [inline lib/Doctrine/Common/Annotations/DocParser.php]: Can we somehow reduce this nesting level?

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: lib/Doctrine/Common/Annotations/DocParser.php (hunk 2 above)
Comment (lyrixx) [reply lib/Doctrine/Common/Annotations/DocParser.php]: done ;)

File: lib/Doctrine/Common/Annotations/DocParser.php (hunk 1 above)
Comment (lyrixx) [reply lib/Doctrine/Common/Annotations/DocParser.php]: oh yes. Fixed

**Review by ammarfaizi2 — DISMISSED** [review]

**Comment by alcaeus** [comment]
@lyrixx sorry for the delay here - I've pinged @Ocramius for another review. In the meantime, could I ask you to rebase your PR?

**Comment by lyrixx** [comment]
I have rebased this PR

**Review by alcaeus — DISMISSED** [review]

**Comment by lyrixx** [comment]
Could we merge this PR because another conflict?
Thanks

**Comment by SenseException** [comment]
@alcaeus Anything we need here before this can be merged?

@lyrixx We need probably another rebase to the latest changes.

**Comment by lyrixx** [comment]
I'm sorry, but I don't understand what is wrong here. I keep rebase my PR but it's never merged.

@Ocramius Can you me help us to move on?

File: tests/Doctrine/Tests/Annotations/DocParserTest.php [hunk 3]
```diff
@@ -1255,6 +1255,21 @@ public function testSetValuesExeption()
         $this->createTestParser()->parse($docblock, 'some class');
     }
 
+    /**
+     * @expectedException \Doctrine\Annotations\AnnotationException
+     * @expectedExceptionMessage [Creation Error] The annotation @SomeAnnotationClassNameWithoutConstructorAndProperties declared on some class does not have a property
+     */
+    public function testSetValuesExeptionWhenNotAvailableProperties()
```
Comment (greg0ire) [inline tests/Doctrine/Tests/Annotations/DocParserTest.php]:
```suggestion
    public function testSetValuesExceptionWhenNotAvailableProperties()
```

File: tests/Doctrine/Tests/Annotations/DocParserTest.php [hunk 4]
```diff
@@ -1255,6 +1255,21 @@ public function testSetValuesExeption()
...
+    {
+        $docblock = <<<DOCBLOCK
+/**
+ * @SomeAnnotationClassNameWithoutConstructorAndProperties(invalidaProperty = "Some val")
```
Comment (greg0ire) [inline tests/Doctrine/Tests/Annotations/DocParserTest.php]:
```suggestion
 * @SomeAnnotationClassNameWithoutConstructorAndProperties(invalidProperty = "Some val")
```

File: tests/Doctrine/Tests/Annotations/DocParserTest.php [hunk 5]
```diff
@@ -1255,6 +1255,21 @@ public function testSetValuesExeption()
         $this->createTestParser()->parse($docblock, 'some class');
     }
 
+    /**
+     * @expectedException \Doctrine\Annotations\AnnotationException
+     * @expectedExceptionMessage [Creation Error] The annotation @SomeAnnotationClassNameWithoutConstructorAndProperties declared on some class does not have a property
```
Comment (guilhermeblanco) [inline tests/Doctrine/Tests/Annotations/DocParserTest.php]: This exception message does not correlate to any of the messages that this patch provides. Maybe you're testing another piece of code, but it seems to me that this test does not cover the added code.

Can you please add the test covering the implemented scenarios?

**Comment by lyrixx** [comment]
Should be OK

**Comment by lyrixx** [comment]
hello @Ocramius @guilhermeblanco 
What is missing?

**Comment by lyrixx** [comment]
The diff is easier to read with [`?=1`](https://github.com/doctrine/annotations/pull/167/files?w=1)

I think it's an easy pick

**Review by greg0ire — APPROVED** [review]

**Review by Ocramius — APPROVED** [review]

**Pull request merged**